src/evm/protocol/uniswap_v4/hooks/hook_handler_creator.rs

```diff
@@ -44,8 +44,62 @@ pub trait HookHandlerCreator: Send + Sync {
     ) -> Result<Box<dyn HookHandler>, InvalidSnapshotError>;
 }
 
+pub struct GenericVMHookHandlerCreator;
+
+impl HookHandlerCreator for GenericVMHookHandlerCreator {
+    fn instantiate_hook_handler(
+        &self,
+        _params: HookCreationParams,
+    ) -> Result<Box<dyn HookHandler>, InvalidSnapshotError> {
+        todo!()
+    }
+}
+
 // Workaround for stateless decoder trait.
+// Mapping from hook address to the handler creator.
 lazy_static! {
     static ref HANDLER_FACTORY: RwLock<HashMap<Address, Box<dyn HookHandlerCreator>>> =
         RwLock::new(HashMap::new());
 }
+
+lazy_static! {
+    static ref DEFAULT_HANDLER: RwLock<Box<dyn HookHandlerCreator>> =
```

Comment:
why does it need a `RwLock`?

Reply:
Will it never be that multiple threads attempt to set the default creator at once? I assumed if that were the case for the factory, that it could also be the case here?